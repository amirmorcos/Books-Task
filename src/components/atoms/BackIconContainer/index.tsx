import BackIcon from "assets/icons/BackIcon";
import { useAppTheme } from "hooks/useAppTheme";
import { useAppNavigation } from "navigation/types";
import React from "react";
import { TouchableOpacity } from "react-native";
import styles from "./styles";
import type { BackIconContainerProps } from "./types";

const BackIconContainer = ({
  overrideContainerStyle,
}: BackIconContainerProps) => {
  const { goBack } = useAppNavigation();
  const { colors } = useAppTheme();

  const themedStyles = styles(colors);
  return (
    <TouchableOpacity
      style={[themedStyles.container, overrideContainerStyle]}
      onPress={goBack}
    >
      <BackIcon />
    </TouchableOpacity>
  );
};

export default BackIconContainer;
