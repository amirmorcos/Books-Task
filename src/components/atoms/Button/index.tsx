import { useAppTheme } from "hooks/useAppTheme";
import React from "react";
import { Text, TouchableOpacity } from "react-native";
import styles from "./styles";
import type { ButtonProps } from "./types";

const Button = ({
  onPress,
  size = "large",
  variant = "primary",
  label,
  overrideContainerStyle,
  overrideTitleStyle,
  disabled,
}: ButtonProps) => {
  const { colors } = useAppTheme();
  const themedStyles = styles(colors);
  const { fontFamily, fontSize, ...sizeStyle } = themedStyles[size];
  const { color, ...variantStyle } = themedStyles[variant];

  return (
    <TouchableOpacity
      disabled={disabled}
      style={[
        themedStyles.container,
        sizeStyle,
        variantStyle,
        disabled && themedStyles.disabled,
        overrideContainerStyle,
      ]}
      onPress={onPress}
    >
      <Text
        style={[
          { color, fontSize, fontFamily },
          disabled && themedStyles.disabledText,
          overrideTitleStyle,
        ]}
      >
        {label}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
