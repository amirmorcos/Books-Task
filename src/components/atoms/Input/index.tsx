import { useAppTheme } from "hooks/useAppTheme";
import React, { useState } from "react";
import {
  NativeSyntheticEvent,
  TextInput,
  TextInputFocusEventData,
  View,
} from "react-native";
import styles from "./styles";
import { InputProps } from "./types";

const Input = ({
  overrideContainerStyle,
  overrideInputStyle,
  leftIcon,
  onFocus,
  onBlur,
  ...props
}: InputProps) => {
  const [isFocused, setIsFocused] = useState(false);
  const { colors } = useAppTheme();
  const themedStyles = styles(colors);

  const onInputFocus = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
    setIsFocused(true);
    onFocus?.(e);
  };

  const onInputBlur = (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
    setIsFocused(false);
    onBlur?.(e);
  };

  return (
    <View
      style={[
        themedStyles.container,
        isFocused && themedStyles.focused,
        overrideContainerStyle,
      ]}
    >
      {leftIcon}
      <TextInput
        onFocus={onInputFocus}
        onBlur={onInputBlur}
        style={[themedStyles.input, overrideInputStyle]}
        {...props}
      />
    </View>
  );
};

export default Input;
