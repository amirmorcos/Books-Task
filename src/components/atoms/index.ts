import Button from "./Button";
import Input from "./Input";
import BackIconContainer from "./BackIconContainer";

export { Button, Input, BackIconContainer };
