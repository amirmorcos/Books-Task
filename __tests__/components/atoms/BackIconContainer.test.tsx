/**
 * @format
 */

import React from "react";

// Note: import explicitly to use the types shipped with jest.

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { BackIconContainer } from "atoms/index";

it("renders correctly", () => {
  const tree = renderer.create(<BackIconContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
