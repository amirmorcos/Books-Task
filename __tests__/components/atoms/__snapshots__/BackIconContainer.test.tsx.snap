// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 30,
      "borderWidth": 1,
      "height": [MockFunction],
      "justifyContent": "center",
      "opacity": 1,
      "width": [MockFunction],
    }
  }
>
  <RNSVGSvgView
    bbHeight={24}
    bbWidth={24}
    fill="none"
    focusable={false}
    height={24}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    width={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M9.57 5.93 3.5 12l6.07 6.07M20.5 12H3.67"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
            "strokeMiterlimit",
          ]
        }
        stroke={
          {
            "payload": 4279374354,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
